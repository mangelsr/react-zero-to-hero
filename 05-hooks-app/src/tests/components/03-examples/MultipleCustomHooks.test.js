import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';

import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useCounter } from '../../../hooks/useCounter';
import { useFetch } from '../../../hooks/useFetch';


jest.mock('../../../hooks/useCounter');
jest.mock('../../../hooks/useFetch');

describe('Tests over MultipleCustomHooks component', () => {
    
    useCounter.mockReturnValue({
        counter: 10,
        increment: () => {}
    });
    
    test('should match snapshot', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should show info', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Miguel',
                quote: 'Hello World'
            }],
            loading: false,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hello World');
        expect(wrapper.find('footer').text().trim()).toBe('Miguel');
    });

});