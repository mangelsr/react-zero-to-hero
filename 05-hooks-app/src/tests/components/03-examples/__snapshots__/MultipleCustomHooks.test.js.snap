// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests over MultipleCustomHooks component should match snapshot 1`] = `
<Fragment>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
</Fragment>
`;
